### PR TITLE
boards: bl652_dvk: Add BL652 DVK board

### PR DESCRIPTION
**Pull request by rerickson1**
Add the Laird Connectivity BL652 DVK board to zephyr.
This board is closely based on the nrf52-pca10040.

Signed-off-by: Ryan Erickson <ryan.erickson@lairdtech.com>

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for docs

File: boards/arm/bl652_dvk/bl652_dvk.dts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+&i2c0 {
+	status = "ok";
+	sda-pin = <26>;
+	scl-pin = <27>;
+};
+
+&pwm0 {
+	status = "ok";
+	ch0-pin = <17>;
+	ch0-inverted;
```
Comment (anangl) [inline boards/arm/bl652_dvk/bl652_dvk.dts]: LED1 is active high, so the channel should not be inverted.

**Comment by rerickson1** [comment]
Just want to confirm if there is anything else needed on this.  The shippable failure looks unrelated to my changes.

File: boards/arm/bl652_dvk/bl652_dvk.dts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+};
+
+&spi0 {
+	status = "ok";
+	sck-pin = <25>;
+	mosi-pin = <23>;
+	miso-pin = <24>;
```
Comment (anangl) [inline boards/arm/bl652_dvk/bl652_dvk.dts]: Wouldn't it be good to add `cs-gpios = <&gpio0 22 0>;` here? And `cs-gpios = <&gpio0 12 0>;` in `spi1` below, as the Dev Kit User Guide mentions these pins?

Comment (rerickson1) [reply boards/arm/bl652_dvk/bl652_dvk.dts]: Good catch. Fixed.

**Comment by tautologyclub** [comment]
Also, 

```
CONFIG_UART_CONSOLE=y
CONFIG_RTT_CONSOLE=y
```

is this kosher?

**Comment by rerickson1** [comment]
> Also,
> 
> ```
> CONFIG_UART_CONSOLE=y
> CONFIG_RTT_CONSOLE=y
> ```
> is this kosher?

I got this from the latest nrf52-pca10040 board file so I assumed it was ok.  The nrf52840-pca10056 also uses it. Maybe @carlescufi can comment.

**Comment by nashif** [comment]
#13335 to fix CI issues.

**Pull request merged**